simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp

```diff
@@ -58,17 +58,15 @@ void SimModelDelaySteerAccGeared::update(const float64_t & dt)
   delayed_input(IDX_U::STEER_DES) = steer_input_queue_.front();
   steer_input_queue_.pop_front();
 
-  const auto prev_vx = state_(IDX::VX);
-
+  const auto prev_state = state_;
   updateRungeKutta(dt, delayed_input);
 
   // take velocity limit explicitly
   state_(IDX::VX) = std::max(-vx_lim_, std::min(state_(IDX::VX), vx_lim_));
 
-  state_(IDX::VX) = calcVelocityWithGear(state_, gear_);
-
-  // calc acc directly after gear consideration
-  state_(IDX::ACCX) = (state_(IDX::VX) - prev_vx) / std::max(dt, 1.0e-5);
+  // consider gear
+  // update position and velocity first, and then acceleration is calculated naturally
```

Comment:
@takayuki5168 I think `a` should be `(v - v_prev) / dt` only when `v=0` is addressed.